### ISSUE TITLE
[lines-around-comment] Does not respect interfaces

### ISSUE THREAD

**Issue by yordis** [typescript-eslint/typescript-eslint#1150]
I am not sure where to open this since the rule is not from `typescript-eslint` but since it only happens with TypeScript interfaces I am looking for direction.

**Repro**

```JSON
{
  "rules": {
    "lines-around-comment": [
      "error",
      {
        "beforeBlockComment": true,
        "afterBlockComment": false,
        "beforeLineComment": true,
        "afterLineComment": false,
        "allowBlockStart": true,
        "allowBlockEnd": false,
        "allowObjectStart": true,
        "allowObjectEnd": false,
        "allowArrayStart": true,
        "allowArrayEnd": false,
        "allowClassStart": true,
        "allowClassEnd": false,
      },
    ],
  }
}
```

```TS
// your repro code case

export interface IStartCommandOptions {
  /**
   * This will throw an error using
   */
  dangerouslyDisableHostCheck: boolean;
}
```

**Expected Result**
Shouldn't fail using interfaces

**Actual Result**

```
error  Expected line before comment  lines-around-comment
```

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.5.0` |
| `@typescript-eslint/parser`        | `2.5.0` |
| `TypeScript`                       | `3.6.4` |
| `ESLint`                           | `6.6.0` |
| `node`                             | `10.16.1` |
| `npm`                              | `6.12.0` |

**Comment by d3lm** [comment]
Any updates on this?

**Comment by yubaoquan** [comment]
Fail on enum either

**Comment by bradrich** [comment]
Interested in the progress of this.

**Comment by bradzacher** [comment]
With any issue in opened in this project - it either has a visible progress in the form of an attached PR, or it has no progress.

We are a community run project. The **volunteer** maintainers spend most of their time triaging issues and reviewing PRs. This means that most issues *will not progress* unless a member of the community steps up and **champions** it.

If this issue is important to you - consider being that champion.

If not - please just subscribe to the issue and wait patiently.
Commenting asking for status updates does not bump issue priority in any way and just serves to spam ***everyone*** subscribed to the issue.

**Comment by bradrich** [comment]
Thank you for that clarification. I apologize and will read your contribution guidelines for future reference.

**Comment by zkldi** [comment]
Is this still accepting PRs? I might be free to tackle this soon-ish.

**Comment by JoshuaKGoldberg** [comment]
Yup, still accepting PRs @zkldi! Go ahead 🙌

**Comment by JesuisTong** [comment]
Anybody works on this issue?
